### DISCUSSION TITLE
What are the supported limits ?

### DISCUSSION THREAD

**Discussion by piochard** [hneemann/Digital#763] [Q&A] [answered]
I have a hierarchical circuit and simulation just hangs while Circuit Statistics just returns nothing after blocking for a while. It sounds like I'm hitting some kind of limits. The circuit should have approximately 132000 NAND and 8000 D FlipFlop.

Simulation I don't much care about, it's probably impractical. Statistics would be nice though.

**Comment by hneemann** [comment, marked as answer]
There is no specific limit. I think it is a problem with the memory consumption. With 100k gates it should still run fine, although a bit laggy. But when the memory gets low, the VM spends a lot of time in the garbage collector, and it looks like the application hangs.

Try to start the simulator using the comand line via

```
java -Xmx8192m -jar Digital.jar
```

This way, the VM can use 8 GBytes, which should be sufficient.

**Reply by piochard** [reply]
Yes, of course!

Both simulation and statistics work fine with -Xms1024m.

**Reply by hneemann** [reply]
I may have been a bit generous with my 8 GBytes advice! :smile:
